src/contracts/AludelFactory.sol

```diff
@@ -207,9 +205,12 @@ contract AludelFactory is Ownable, InstanceRegistry {
         external
         onlyOwner
     {
-        // register aludel instance
-        // if program is already registered this will revert
-        _register(program);
+        if(isAludel(program)){
+          revert AludelAlreadyRegistered();
+        }
+        if (template == address(0)) {
+            revert InvalidTemplate();
+        }
```

Comment:
why not check if the template is registered as well?

Reply:
Actually, I think we need to drop this condition. Because we'll have to migrate current existing programs and register them in the factory. These programs won't have a template to match with. We could assign them a template with the same ABI for the sake of clarity but I don't know if it's worth it.

Reply:
> 
I don’t believe that’s correct, the frontend will need to identify the program type by template

Reply:
Assigning it a template with the same ABI will be what’s important here

Reply:
Thank you for clarifying this @joehquak!

Reply:
Another reason to use a template other than zero is to use the template field to know if the storage slot for the struct is used or not, and know for sure if an address is a template. Otherwise we could use another field, or add a boolean for it in the struct.